### PR TITLE
CMake: Make it easier to find boost and qt on windows

### LINKED ISSUE #3529: Improvements to be done for CGAL CMake scripts
## Issue Details

@afabri and [Max Lyon] (not on Github) showed me how hard it can be to configure the polyhedron demo of CGAL, on Windows.

[Max Lyon]: https://www.graphics.rwth-aachen.de/person/99/

### Various issues
- [x] `FindTBB` displays a message starting with `ERROR: `, for messages that are just warnings. It should check the variables `TBB_FIND_REQUIRED` and `TBB_FIND_QUIET`, and display messages accordingly.
- [x] Qt5 is hard to configure. If CMake does not auto-detect the install of Qt5, that is rather difficult to understand how should the CMake variable `Qt5_DIR` be filled. It is not the root of the installation of Qt5 on Windows.
- [ ] GMP/MPFR: if one uses a version of CGAL installed by the CGAL Windows installer, that is fine, but as soon as you use a git checkout of CGAL, things get complicated.
- [x] Boost... Boost installation layout is strange, on Windows. It is not strait forward to know what to fill in CMake.

### Various proposals

- All CGAL `CMakeLists.txt` should start with `find_package(CGAL REQUIRED)`, before any other `find_package`, so that CMake forces the users to first configure the path to `CGAL_DIR`. That would ensure that `CGALConfig.cmake` (and all the CGAL CMake machinery) is already included for all the remaining lines of the `CMakeLists.txt`.
- That way, we could fill `CMAKE_PREFIX_PATH` with the known standard installation paths for Qt5 and Booost.
- I also suggest that the `else()` sections of our `CMakeLists.txt` explain how to configure the missing dependencies. That could probably be easily done with the usage of the [`FeatureSummary`] module of CMake. OpenFlipper [already uses it](https://bitbucket.org/fenics-project/dolfin/pull-requests/275/use-cmakes-featuresummary-for-generating-a/diff).

[`FeatureSummary`]: https://cmake.org/cmake/help/v3.13/module/FeatureSummary.html

### PR DESCRIPTION
**Pull request by maxGimeno**
## Summary of Changes

Add paths to Qt5 and boost windows default installation directories to CMAKE_PREFIX_PATH in CMakeLists.

## Release Management

* Affected package(s):CGAL
* Issue(s) solved (if any): fix #3529

### CODE REVIEW

**Comment by janetournois** [comment]
May I ask why you are using `BOOST_DIRS `?

[CMake documentation](https://cmake.org/cmake/help/latest/module/FindBoost.html) mentions `BOOST_ROOT` and [CGAL documentation](https://doc.cgal.org/latest/Manual/windows.html#title9) mentions `BOOST_LIBRARYDIR` and `BOOST_INCLUDEDIR`.

On my windows setup, I have the environment variable `BOOST_ROOT` which is enough to find everything.

**Comment by maxGimeno** [comment]
@janetournois BOOST_DIRS is just a simple variable to store the list. The goal here is to detect if a directory is at the default location on windows so the user doesn't have to export BOOST_ROOT in a standard installation case.

File: CMakeLists.txt [hunk 1]
```diff
@@ -57,6 +57,27 @@ if(BUILD_TESTING)
...
+  list(APPEND CMAKE_PREFIX_PATH "${boost_dir}")
+endif()#BOOST_DIRS
+
+file(GLOB QT_DIRS "C:/Qt/5")
+if(QT_DIRS)
+  list(GET QT_DIRS 0 qt_dir)
+  file(GLOB compilos "${qt_dir}/msvc*")
+  if(compilos)
+    list(GET compilos 0 compilo)
+    list(APPEND CMAKE_PREFIX_PATH "${compilo}/lib/cmake/Qt5")
```
Comment (lrineau) [inline CMakeLists.txt:76]: `/lib/cmake/Qt5` should not be part of `CMAKE_PREFIX_PATH`. The command `find_package` already knows where to search for in the prefix. For example, on certain 64 bits Linux systems, the path is `/lib64/cmake/Qt5`.

```suggestion
    list(APPEND CMAKE_PREFIX_PATH "${compilo}")
```

**Comment by janetournois** [comment]
> 
> 
> @janetournois BOOST_DIRS is just a simple variable to store the list. The goal here is to detect if a directory is at the default location on windows so the user doesn't have to export BOOST_ROOT in a standard installation case.

Ok I understand.
Then, you should add `C:\Program Files (x86)` to the places to be checked.

File: CMakeLists.txt (hunk 1 above)
Comment (afabri) [reply CMakeLists.txt:76]: Really "compilo" ????

**Comment by afabri** [comment]
Did anybody of you install a pre-compiled version of boost?  The installer proposes `C:\local` as directory, and I would assume that at least some people accept what an installer proposes.
image

**Comment by maxGimeno** [comment]
That's the information I was lookin for, but I only found Program Files...

**Comment by maxGimeno** [comment]
Successfully tested in https://cgal.geometryfactory.com/CGAL/testsuite/results-5.3-Ic-34.shtml

**Pull request merged**